presto-parquet/src/main/java/com/facebook/presto/parquet/batchreader/decoders/delta/FixedLenByteArrayShortDecimalDeltaValueDecoder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.batchreader.decoders.delta;
+
+import com.facebook.presto.common.type.Decimals;
+import com.facebook.presto.parquet.batchreader.decoders.ValuesDecoder.ShortDecimalValuesDecoder;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.values.ValuesReader;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.LogicalTypeAnnotation.DecimalLogicalTypeAnnotation;
+import org.openjdk.jol.info.ClassLayout;
+
+import static com.facebook.presto.parquet.ParquetTypeUtils.getShortDecimalValue;
+import static com.facebook.presto.parquet.batchreader.decoders.plain.FixedLenByteArrayShortDecimalPlainValuesDecoder.checkBytesFitInShortDecimal;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Note: this is not an optimized values decoder. It makes use of the existing Parquet decoder. Given that this type encoding
+ * is not a common one, just use the existing one provided by Parquet library and add a wrapper around it that satisfies the
+ * {@link ShortDecimalValuesDecoder} interface.
+ */
+public class FixedLenByteArrayShortDecimalDeltaValueDecoder
```

Comment:
avoid abbreviations; thus FixedLen --> FixedLength

but aren't all arrays in java fixed length? So maybe just ByteArrayShortDecimalDeltaValueDecoder

Reply:
The naming convention of the Decoder class name here is ParquetPrimitiveType + [Short|Long]Decimal + encoding + ValuesDecoder. The PrimitiveTypeName corresponding to this class is FIXED_LEN_BYTE_ARRAY, so this name is used.